go-runtime/ftl/pubsub.go

```diff
@@ -3,32 +3,38 @@ package ftl
 import (
 	"context"
 
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/go-runtime/ftl/reflection"
 	"github.com/TBD54566975/ftl/go-runtime/internal"
 )
 
 // Topic declares a topic
 //
 // Topics publish events, and subscriptions can listen to them.
 func Topic[E any](name string) TopicHandle[E] {
-	return TopicHandle[E]{name: name}
+	return TopicHandle[E]{Ref: &schema.Ref{
+		Name:   name,
+		Module: reflection.Module(),
+	}}
 }
 
 type TopicHandle[E any] struct {
-	name string
+	Ref *schema.Ref
 }
 
 // Publish publishes an event to a topic
 func (t TopicHandle[E]) Publish(ctx context.Context, event E) error {
-	return internal.FromContext(ctx).PublishEvent(ctx, t.name, event)
+	return internal.FromContext(ctx).PublishEvent(ctx, t.Ref, event)
+}
+
+type SubscriptionHandle[E any] struct {
+	Topic *schema.Ref
```

Comment:
TopicHandle and SubscriptionHandle's fields are accessed by ftltest, so have been made public